src/lib/sap_ha/system/network.rb

```diff
@@ -41,15 +41,15 @@ def interfaces
       # Get local machine's IPv4 addresses excluding the loopback iface
       def ip_addresses
         interfaces = Socket.getifaddrs.select do |iface|
-          iface.addr.ipv4? && !iface.addr.ipv4_loopback?
+	  !iface.addr.nil? && iface.addr.ipv4? && !iface.addr.ipv4_loopback?
```

Comment:
It is enought to say `iface.addr && iface.addr.ipv4? && !iface.addr.ipv4_loopback?`.